service/src/main/java/bio/terra/pipelines/app/controller/AdminApiController.java

```diff
@@ -74,6 +91,28 @@ public ResponseEntity<ApiAdminPipeline> updatePipeline(
     return new ResponseEntity<>(pipelineToApiAdminPipeline(updatedPipeline), HttpStatus.OK);
   }
 
+  @Override
+  public ResponseEntity<ApiAdminQuota> updateQuotaLimit(
+      String pipelineName, String userId, ApiUpdateQuotaLimitRequestBody body) {
+    final SamUser authedUser = getAuthenticatedInfo();
+    samService.checkAdminAuthz(authedUser);
+    PipelinesEnum validatedPipelineName =
+        PipelineApiUtils.validatePipelineName(pipelineName, logger);
+
+    // Check if a row exists for this user + pipeline. Throw an error if it doesn't
+    // A row should exist if a user has run into a quota issue before.
+    if (!quotasService.isUserQuotaPresent(userId, validatedPipelineName)) {
```

Comment:
curious about the thought process behind this approach, versus adding an option to `getQuotaForUserAndPipeline` that specifies whether to create a new row if the user isn't found and just reusing that method below?

Reply:
hmmm well calling the other admin endpoint will actually create that row for the user cuz of how the`getQuotaForUserAndPipeline` function works so this is all kind of moot.  What scenario do we actually want to catch here?  Is it that the user has used our service before? is it that the user_id is an actual terra user_id?

we could write a get function that errors out if the row doesnt exist instead of generating a new one for these admin calls

Reply:
yeah I'm saying we could add an input to `getQuotaForUserAndPipeline` that's like `addRowIfAbsent=True` and toggle that functionality so we could reuse it, if that input is False and there's no row for that user then we throw. if that feels too complicated, that's fine, just an idea that I'm not at all tied to